pipelines/batch/src/main/java/org/openmrs/analytics/FhirEtl.java

```diff
@@ -211,24 +212,45 @@ static void validateOptions(FhirEtlOptions options) {
 	 */
 	static void runHapiJdbcFetch(FhirEtlOptions options, DatabaseConfiguration dbConfig, FhirContext fhirContext)
 	        throws PropertyVetoException {
-		FhirSearchUtil fhirSearchUtil = createFhirSearchUtil(options, fhirContext);
 		JdbcConnectionUtil jdbcConnectionUtil = new JdbcConnectionUtil(options.getJdbcDriverClass(),
 		        dbConfig.makeJdbsUrlFromConfig(), dbConfig.getDbUser(), dbConfig.getDbPassword(),
 		        options.getJdbcInitialPoolSize(), options.getJdbcMaxPoolSize());
-		JdbcFetchUtil jdbcFetchUtil = new JdbcFetchUtil(jdbcConnectionUtil);
-		Pipeline pipeline = Pipeline.create(options);
+		FhirSearchUtil fhirSearchUtil = createFhirSearchUtil(options, fhirContext);
 		
-		String searchUrl = options.getFhirServerUrl() + "/$get-resource-counts";
-		HashSet<String> resourceTypes = new HashSet<String>(Arrays.asList(options.getResourceList().split(",")));
-		//Get the resource count for each resource type and distribute the query workload evenly across allocated workers.
-		HashMap<String, Integer> resourceCount = fhirSearchUtil.searchResourceCounts(searchUrl, resourceTypes);
-		PCollection<List<String>> payload = jdbcFetchUtil.runJdbcIoFetch(pipeline, options.getResourceList(), resourceCount,
-		    options.getJdbcMaxPoolSize());
-		payload.apply("Convert to parquet", ParDo.of(new ConvertResourceFn(options)));
+		//Get the resource count for each resource type and distribute the query workload based on batch size.
+		HashMap<String, Integer> resourceCount = fhirSearchUtil.searchResourceCounts(options.getResourceList());
+		List<MetricResults> metricResultsList = new ArrayList<MetricResults>();
 		
-		PipelineResult result = pipeline.run();
-		result.waitUntilFinish();
-		EtlUtils.logMetrics(result.metrics());
+		//Run the ETL process for each resource specified in the pipeline options
+		for (String resourceType : options.getResourceList().split(",")) {
+			int numResources = resourceCount.get(resourceType);
+			int numBatches = numResources / options.getHapiJdbcBatchSize();
+			if (numResources % options.getHapiJdbcBatchSize() != 0) {
+				numBatches += 1;
+			}
+			//Run the ETL process for each batch in sequence
+			for (int i = 0; i < numBatches; i++) {
+				Pipeline pipeline = Pipeline.create(options);
```

Comment:
So as discussed offline, this is the main issue that I see with your recent change. We should not need multiple `Pipeline` instances to fix the memory problem you had. Let's wait and see how we can solve that problem and then I will review the rest again.